ludwig/schema/features/base.py

```diff
@@ -1,7 +1,57 @@
+from typing import List
+
+from marshmallow_dataclass import dataclass
+
+from ludwig.constants import (
+    AUDIO,
+    BAG,
+    BINARY,
+    CATEGORY,
+    DATE,
+    H3,
+    IMAGE,
+    NUMBER,
+    SEQUENCE,
+    SET,
+    TEXT,
+    TIMESERIES,
+    VECTOR,
+)
 from ludwig.schema import utils as schema_utils
+from ludwig.schema.metadata.parameter_metadata import ParameterMetadata
+
+
+@dataclass
+class BaseFeatureConfig(schema_utils.BaseMarshmallowConfig):
+    """Base class for feature configs."""
+
+    name: str = schema_utils.String(
+        allow_none=True,
```

Comment:
Shouldn't allow none for `name` and `type` be `false`?

Reply:
If allow_none is false, then we must provide a (non-None) default, so I left allow_none True.  What would be the default if allow_none False?

Reply:
This is a good point. I realized that since the json schema mapping still has them as required params, this won't cause an issue with users potentially not specifying a name or feature type so we can disregard this!